### PR TITLE
feat: Change title for each page, read content of new page

### LINKED ISSUE #200: Accessibility: Update <title> for each page
## Description
Each page should update the <title> tag value with the current page title

### LINKED ISSUE #208: Accessibility: Let users know when page has change
## Description
Right now when the user navigate to different pages there is no way to know something has changed. This is an opportunity tu use aria-live

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Update page title on change route
- Read content when route changes

### Checklist
- [x] Corresponding issue has been opened
- [ ] New tests added
- [x] Strings are localized
- [x] Verified on mobile
- [x] Verified on desktop
- [x] Verified accessibility ([Accessibility - Web applications](https://docs.google.com/document/d/1VwPDyLqw7r_iISMgDZr1FdAmAqqOIIVIxQqsYA-HaHE/edit?usp=sharing))

### Related Issues
Fixes #200
Fixes #208

### CODE REVIEW

**Review by paulschreiber — CHANGES_REQUESTED** [review]
issue: 
The `<title>` tag shouldn't match the `<h1>`.

Here are some titles we can use:

* home page should be `Terraso | Home`
* groups should be `Terraso | Group | Blah` (instead of `Terraso | Blah`). 
* landscapes should be `Terraso | Landscape | Blah` (instead of `Terraso | Blah`). 
* `Terraso | ACUS Penninsula Galera-San Francisco Members` should be `Terraso | Group | ACUS Penninsula Galera-San Francisco | Members`
* `Terraso | Manage "sample" Group Information` should be `Terraso | Group | Sample | Manage

For groups with long names, like `ACUS Peninsula Galera-San Francisco`, we should truncate them in the title … not sure how, though. @Hanal-code What do you recommend?

**Comment by paulschreiber** [comment]
Terraso | Home 
Terraso | Groups
Terraso | Group | Blah | Edit
Terraso | Create Group
Terraso | Group | Blah | Members
Terraso | Group | Blah
Terraso | Landscapes
Terraso | Landscape | Blah | Members
Terraso | Landscape | Blah
Terraso | Tools
Terraso | Log in
Terraso | Account

**Comment by Hanal-code** [comment]
I'd suggest to put the current page title first instead of 'Terraso' to provide descriptive titles based on [W3C document](https://www.w3.org/WAI/WCAG21/Techniques/general/G88) 
When many tabs open, users will see 'Terraso | ...' only which already indicated by favicon. 

The current-page-first approach will make the page title;
Home | Terraso 
Groups  | Terraso 
Edit: Blah | Terraso
Create Group | Terraso
Members: Blah  | Terraso
Blah | Terraso
Landscapes | Terraso
Tools | Terraso
Log in | Terraso
Account | Terraso

I don't recommend to use the page title like a breadcrumb which will be added to the app later on.

**Comment by paulschreiber** [comment]
I like @Hanal-code 's suggestions! Those are an improvement. One change I would make is for the home page to lead with `Terraso` — perhaps just `Terraso` or `Terraso Home`?

**Comment by josebui** [comment]
Thanks @Hanal-code @paulschreiber . Hana let me know what do you think about Paul's suggestion for the home title

**Comment by Hanal-code** [comment]
@paulschreiber  Can you elaborate your rationale to deviate the pattern for the Home page?

**Comment by paulschreiber** [comment]
@Hanal-code starting with `Terraso` allows a good title for typeahead completion and bookmarks.

File: src/common/document.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
-    : `${title} | ${t('common.terraso_projectName')}`;
+  const titleParts = [
+    fetching ? null : title,
+    omitSuffix ? null : t('common.terraso_projectName'),
+  ];
+
+  const fullTitle = _.flow(
+    _.filter(part => !_.isEmpty(part)),
+    _.join(' | ')
+  )(titleParts);
```
Comment (paulschreiber) [inline src/common/document.js:16]: suggestion: simplify this as `_,compact(titleParts).join(' | ')`.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: src/common/document.js (hunk 1 above)
Comment (josebui) [reply src/common/document.js:16]: Thanks, didn't know about that compact function

**Review by paulschreiber — CHANGES_REQUESTED** [review]
issue: when changing pages, the title changes to `Terraso` for a brief period. Please eliminate that extra change.

**Comment by josebui** [comment]
@paulschreiber That happens because the app is loading data, what should be the title while the app doesn't have the information to display the correct title?

File: src/common/document.js (hunk 1 above)
Comment (paulschreiber) [reply src/common/document.js]: Any time you are filtering on "not empty" in a verbose way, it's a hint to look for something briefer. :)

In regular JavaScript, you can also do `foo.filter(bar => bar)` and `foo.filter(Number)`.

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**